instrumentation/jdbc/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/jdbc/test/JdbcInstrumentationTest.java

```diff
@@ -1632,4 +1632,146 @@ void testPreparedBatch(String system, Connection connection, String username, St
+      throws SQLException {
+
+    String tableName = "TXN_ROLLBACK_TEST_" + system.toUpperCase(Locale.ROOT);
+    Statement createTable = connection.createStatement();
+    createTable.execute("CREATE TABLE " + tableName + " (id INTEGER not NULL, PRIMARY KEY ( id ))");
+    cleanup.deferCleanup(createTable);
+
+    boolean originalAutoCommit = connection.getAutoCommit();
+    connection.setAutoCommit(false);
+
+    testing.waitForTraces(1);
+    testing.clearData();
+
+    try {
+      Statement insertStatement = connection.createStatement();
+      cleanup.deferCleanup(insertStatement);
+
+      testing.runWithSpan(
+          "parent",
+          () -> {
+            insertStatement.executeUpdate("INSERT INTO " + tableName + " VALUES(1)");
+            connection.rollback();
+          });
+
+      testing.waitAndAssertTraces(
+          trace ->
+              trace.hasSpansSatisfyingExactly(
+                  span -> span.hasName("parent").hasKind(SpanKind.INTERNAL).hasNoParent(),
+                  span ->
+                      span.hasName("INSERT jdbcunittest." + tableName)
+                          .hasKind(SpanKind.CLIENT)
+                          .hasParent(trace.getSpan(0))
+                          .hasAttributesSatisfyingExactly(
+                              equalTo(maybeStable(DB_SYSTEM), maybeStableDbSystemName(system)),
+                              equalTo(maybeStable(DB_NAME), dbNameLower),
+                              equalTo(DB_USER, emitStableDatabaseSemconv() ? null : username),
+                              equalTo(
+                                  DB_CONNECTION_STRING, emitStableDatabaseSemconv() ? null : url),
+                              equalTo(
+                                  maybeStable(DB_STATEMENT),
+                                  "INSERT INTO " + tableName + " VALUES(?)"),
+                              equalTo(maybeStable(DB_OPERATION), "INSERT"),
+                              equalTo(maybeStable(DB_SQL_TABLE), tableName)),
+                  span ->
+                      span.hasName("ROLLBACK")
+                          .hasKind(SpanKind.CLIENT)
+                          .hasParent(trace.getSpan(0))
+                          .hasAttributesSatisfyingExactly(
+                              equalTo(maybeStable(DB_SYSTEM), maybeStableDbSystemName(system)),
+                              equalTo(maybeStable(DB_NAME), dbNameLower),
+                              equalTo(DB_USER, emitStableDatabaseSemconv() ? null : username),
+                              equalTo(
+                                  DB_CONNECTION_STRING,
+                                  emitStableDatabaseSemconv() ? null : url))));
+
+      Statement selectStatement = connection.createStatement();
+      cleanup.deferCleanup(selectStatement);
+      ResultSet resultSet = selectStatement.executeQuery("SELECT COUNT(*) FROM " + tableName);
+      resultSet.next();
+      assertThat(resultSet.getInt(1)).isEqualTo(0);
```

Comment:
I suspect that this could cause test flakiness. `executeQuery` creates a span but since it is not asserted there is a possibility that the span arrives after the test has completed and exported data is reset before the next test. If that happens the next test will fail because there is an unexpected span. You could work around this by calling `testing.clearData();` before this code (to clear the existing trace) and `testing.waitForTraces(1);` after this code to ensure that trace data gets cleared before the next test.